modules/gapi/samples/face_detection.cpp

```diff
@@ -0,0 +1,764 @@
+            faces[i].bbox.y2 = faces[i].bbox.y2 + static_cast<double>(faces[i].regression[2]) * bboxHeight;
+        }
+    }
+
+    static void bboxes2Squares(std::vector<Face>& faces) {
+        for (size_t i = 0; i < faces.size(); ++i) {
+            faces[i].bbox = faces[i].bbox.getSquare();
+        }
+    }
+
+    static std::vector<Face> runNMS(std::vector<Face>& faces, double threshold,
+                                    bool useMin = false) {
+        std::vector<Face> facesNMS;
+        if (faces.empty()) {
+            return facesNMS;
+        }
+
+        std::sort(faces.begin(), faces.end(), [](const Face& f1, const Face& f2) {
+            return f1.score > f2.score;
+        });
+
+        std::vector<int> indices(faces.size());
+        std::iota(indices.begin(), indices.end(), 0);
+
+        while (indices.size() > 0) {
+            const int idx = indices[0];
+            facesNMS.push_back(faces[idx]);
+            std::vector<int> tmpIndices = indices;
+            indices.clear();
+            for (size_t i = 1; i < tmpIndices.size(); ++i) {
+                int tmpIdx = tmpIndices[i];
+                const double interX1 = std::max(faces[idx].bbox.x1, faces[tmpIdx].bbox.x1);
+                const double interY1 = std::max(faces[idx].bbox.y1, faces[tmpIdx].bbox.y1);
+                const double interX2 = std::min(faces[idx].bbox.x2, faces[tmpIdx].bbox.x2);
+                const double interY2 = std::min(faces[idx].bbox.y2, faces[tmpIdx].bbox.y2);
+
+                const double bboxWidth = std::max(0.0, (interX2 - interX1 + 1));
+                const double bboxHeight = std::max(0.0, (interY2 - interY1 + 1));
+
+                const double interArea = bboxWidth * bboxHeight;
+                // TODO: compute outside the loop
+                const double area1 = (faces[idx].bbox.x2 - faces[idx].bbox.x1 + 1) *
+                    (faces[idx].bbox.y2 - faces[idx].bbox.y1 + 1);
+                const double area2 = (faces[tmpIdx].bbox.x2 - faces[tmpIdx].bbox.x1 + 1) *
+                    (faces[tmpIdx].bbox.y2 - faces[tmpIdx].bbox.y1 + 1);
+                double overlap = 0.0;
+                if (useMin) {
+                    overlap = interArea / std::min(area1, area2);
+                }
+                else {
+                    overlap = interArea / (area1 + area2 - interArea);
+                }
+                if (overlap <= threshold) {
+                    indices.push_back(tmpIdx);
+                }
+            }
+        }
+        return facesNMS;
+    }
+};
+
+const double P_NET_WINDOW_SIZE = 12.0;
+const double P_NET_STRIDE = 2.0;
+
+std::vector<Face> buildFaces(const cv::Mat& scores,
+    const cv::Mat& regressions,
+    const double scaleFactor,
+    const double threshold) {
+
+    auto w = scores.size[3];
+    auto h = scores.size[2];
+    auto size = w * h;
+
+    const float* scores_data = scores.ptr<float>();
+    scores_data += size;
```

Comment:
Why did the pointer move to h*w? What about other scores?

Reply:
This is how it works. Scores are actually pairs scores_data[0] + scores_data[size] = 1.0f; We care about second part of the blob. scores_data[size .... 2*size-1] the larger is belter. The larger means is more face-like BB. Again this code is mostly copy-pasted